code/modules/research/nanites/nanite_programs/utility.dm

```diff
@@ -258,6 +258,33 @@
 			SEND_SIGNAL(infectee, COMSIG_NANITE_SYNC, nanites)
 			infectee.investigate_log("was infected by spreading nanites by [key_name(host_mob)] at [AREACOORD(infectee)].", INVESTIGATE_NANITES)
 
+/datum/nanite_program/triggered/nanite_sting
+	name = "Nanite Sting"
+	desc = "Stings a random non-host around the host with a barely-visible cluster of nanites, making them a new host. The target will feel it. \
+			If the cluster finds no valid targets, it returns to the original host."
+	trigger_cost = 5
+	trigger_cooldown = 100
+	rogue_types = list(/datum/nanite_program/glitch, /datum/nanite_program/toxic)
+
+/datum/nanite_program/triggered/nanite_sting/trigger()
+	if(!..())
+		return
+	var/list/mob/living/target_hosts = list()
+	for(var/mob/living/L in oview(1, host_mob))
```

Comment:
code-wise this looks ok, only part i'm kinda scared about since it might get called quite a bit. I don't think it'll be too bad with just 1range but 🤷‍♂

Reply:
Keep in mind that the program has a cooldown. Shouldn't be that bad, right?

Reply:
depends on how many people are infected right? you send the signal and everyone with the bots does the thing?

Reply:
Yeah, but it's not like everyone's gonna be nearby many people all the time. Especially since there will definitely be people that'll notice the prick notification and will start yelling "ling!"

Reply:
Check Adjacent, or this will be able to infect through window panes

Reply:
Done.